policy/approval/approve.go

```diff
@@ -275,16 +276,23 @@ func (r *Rule) filteredCandidates(ctx context.Context, prctx pull.Context) ([]*c
 
 	var allowedCandidates []*common.Candidate
 	for _, candidate := range candidates {
-		if candidate.CreatedAt.After(*last.PushedAt) {
-			allowedCandidates = append(allowedCandidates, candidate)
+		if r.Options.IgnoreEditedComments {
+			if candidate.UpdatedAt != candidate.CreatedAt {
+				log.Debug().Msg("discarded candidate because it was an edited comment")
+				continue
+			}
+		}
+		if r.Options.InvalidateOnPush {
+			if candidate.CreatedAt.Before(*last.PushedAt) {
+				log.Debug().Msgf("discarded candidate because it was invalidated by push of %s at %s",
+					last.SHA,
+					last.PushedAt.Format(time.RFC3339))
+				continue
+			}
 		}
+		allowedCandidates = append(allowedCandidates, candidate)
```

Comment:
@bluekeyes like this? or were you thinking something different?

Reply:
Yeah, this is the right direction, but I think it needs a few more adjustments.